lib/srv/db/sqlserver/connect.go

```diff
@@ -64,20 +68,42 @@ func (c *connector) Connect(ctx context.Context, sessionCtx *common.Session, log
 		TypeFlags:    loginPacket.TypeFlags(),
 	}
 
-	auth, err := c.getAuth(sessionCtx)
-	if err != nil {
-		return nil, nil, trace.Wrap(err)
-	}
-
-	connector := mssql.NewConnectorConfig(msdsn.Config{
+	dsnConfig := msdsn.Config{
 		Host:         host,
 		Port:         portI,
 		Database:     sessionCtx.DatabaseName,
 		LoginOptions: options,
 		Encryption:   msdsn.EncryptionRequired,
 		TLSConfig:    tlsConfig,
 		PacketSize:   loginPacket.PacketSize(),
-	}, auth)
+	}
+
+	var connector *mssql.Connector
+	switch {
+	case sessionCtx.Database.IsAzure() && sessionCtx.Database.GetAD().IsEmpty():
+		// If the client is connecting to Azure SQL, and no AD configuration is
+		// provided, authenticate using the Azure AD Integrated authentication
+		// method.
+		managedIdentityID, err := c.Auth.GetAzureIdentityResourceID(ctx, sessionCtx.DatabaseUser)
+		if err != nil {
+			return nil, nil, trace.Wrap(err)
+		}
+
+		connector, err = azuread.NewConnectorFromConfig(dsnConfig, map[string]string{
+			"fedauth":     azuread.ActiveDirectoryManagedIdentity,
+			"resource id": managedIdentityID,
+		})
+		if err != nil {
+			return nil, nil, trace.Wrap(err)
+		}
+	default:
+		auth, err := c.getAuth(sessionCtx)
```

Comment:
If we're using Azure AD integration, there is no `auth`, and we change the connector (not the auth itself), so we don't need the `aadAuth`. We could move the "selection logic" to `auth` but I'm not sure if that would make part of auth or connection. For me, it makes more sense to be here (at `connect.go`) since we're not directly dealing with auth when Azure AD is being used.